src/vm/functions/conversions.rs

```diff
@@ -113,3 +115,107 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            let as_string = String::from_utf8(data).unwrap();
+            return conversion_fn(as_string);
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flat = data.into_iter().flatten().collect();
+            let as_string = String::from_utf8(flat).unwrap();
+            return conversion_fn(as_string);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![
+                    TypeSignature::max_string_ascii(),
+                    TypeSignature::max_string_utf8(),
+                ],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    fn convert_to_int(raw_string: String) -> Result<Value> {
```

Comment:
Style nit -- we don't have nested functions.  Please either make this into a top-level function, or turn it into an anonymous function argument to `native_string_to_int_generic()`.  Same goes for `native_string_to_uint` below.

Reply:
I removed the nested functions from this change.